### PR TITLE
[SOL-1684] Remove coupon tests dependence on the migrations

### PR DESCRIPTION
**Pull request by marjev**
JIRA ticket: [SOL-1684](https://openedx.atlassian.net/browse/SOL-1684)

1st review: tagging both @mjfrey and @vkaracic. Please take a look.

2nd review: @rlucioni, I would like you to take a look as well since you reported the issue.

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @marjev! It looks like you're a member of a company that does contract work for edX. If you're doing this work as part of a paid contract with edX, you should talk to edX about who will review this pull request. If this work is not part of a paid contract with edX, then you should ensure that there is an OSPR issue to track this work in [JIRA](https://openedx.atlassian.net), so that we don't lose track of your pull request.

To automatically create an OSPR issue for this pull request, just visit this link: https://openedx-webhooks.herokuapp.com/github/process_pr?repo=edx%2Fecommerce&number=604

**Comment by vkaracic** [comment]
The text in the ticket is a bit vague, is this PR regarding the coupons and basket tests or all tests? Because when I run `make validate` with `DISABLE_MIGRATIONS=1` there are a few tests failing, mostly regarding payment processor tests.

File: ecommerce/courses/tests/factories.py [hunk 1]
```diff
@@ -12,3 +18,13 @@ class Meta(object):
 
     id = FuzzyText(prefix='course-id-')
     name = FuzzyText(prefix='course-name-')
+
+    @classmethod
+    def create(cls, **kwargs):
+        if os.getenv('DISABLE_MIGRATIONS'):
```
Comment (clintonb) [inline ecommerce/courses/tests/factories.py]: This is not necessary. [CourseCatalogTestMixin](https://github.com/edx/ecommerce/blob/ff2c224ece949a4d85c41f34235b97ec0c068946/ecommerce/extensions/catalogue/tests/mixins.py#L14) does this for you.

File: ecommerce/tests/mixins.py [hunk 2]
```diff
@@ -298,7 +302,13 @@ class CouponMixin(object):
     """ Mixin for preparing data for coupons and creating coupons. """
     def setUp(self):
         super(CouponMixin, self).setUp()
-        self.category = create_from_breadcrumbs('Coupons > Test category')
+        self.category = factories.CategoryFactory()
+
+        if os.getenv('DISABLE_MIGRATIONS'):
```
Comment (clintonb) [inline ecommerce/tests/mixins.py]: Remove. This mixin has nothing to do with course seats, and should focus only on coupons. Tests that need to test course seats should inherit [CourseCatalogTestMixin](https://github.com/edx/ecommerce/blob/ff2c224ece949a4d85c41f34235b97ec0c068946/ecommerce/extensions/catalogue/tests/mixins.py#L14).

File: ecommerce/tests/mixins.py [hunk 3]
```diff
@@ -298,7 +302,13 @@ class CouponMixin(object):
...
+            get_or_create_seat_product_class()
+            get_or_create_seat_category()
+            coupon = get_or_create_coupon_product_class()
```
Comment (clintonb) [inline ecommerce/tests/mixins.py]: Follow the pattern set by [CourseCatalogTestMixin](https://github.com/edx/ecommerce/blob/ff2c224ece949a4d85c41f34235b97ec0c068946/ecommerce/extensions/catalogue/tests/mixins.py#L14). Stop using `if os.getenv('DISABLE_MIGRATIONS')`. Your migrations should be smart enough to detect the existing data. If they aren't, update them.

**Comment by marjev** [comment]
@vkaracic I fixed the coupon tests since I'm familiar only with that part of the code.

File: ecommerce/courses/tests/factories.py (hunk 1 above)
Comment (marjev) [inline ecommerce/courses/tests/factories.py]: Done.

File: ecommerce/tests/mixins.py (hunk 2 above)
Comment (marjev) [inline ecommerce/tests/mixins.py]: Done.

File: ecommerce/tests/mixins.py (hunk 3 above)
Comment (marjev) [inline ecommerce/tests/mixins.py]: Done.

**Comment by rlucioni** [comment]
Thanks @marjev. This resolves the issues I was seeing with the coupon tests. :+1:

**Comment by vkaracic** [comment]
:+1:

**Pull request merged**